examples/keyvalue-inmemory/main.go

```diff
@@ -19,22 +19,38 @@ func main() {
 }
 
 func run() error {
+	handler := Handler{}
 	p, err := provider.New(
-		provider.SourceLinkPut(handleNewSourceLink),
-		provider.TargetLinkPut(handleNewTargetLink),
-		provider.SourceLinkDel(handleDelSourceLink),
-		provider.TargetLinkDel(handleDelTargetLink),
-		provider.HealthCheck(handleHealthCheck),
-		provider.Shutdown(handleShutdown),
+		provider.SourceLinkPut(func(link provider.InterfaceLinkDefinition) error {
+			return handleNewSourceLink(&handler, link)
+		}),
+		provider.TargetLinkPut(func(link provider.InterfaceLinkDefinition) error {
+			return handleNewTargetLink(&handler, link)
+		}),
+		provider.SourceLinkDel(func(link provider.InterfaceLinkDefinition) error {
+			return handleDelSourceLink(&handler, link)
+		}),
+		provider.TargetLinkDel(func(link provider.InterfaceLinkDefinition) error {
+			return handleDelTargetLink(&handler, link)
+		}),
+		provider.HealthCheck(func() string {
+			return handleHealthCheck(&handler)
+		}),
+		provider.Shutdown(func() error {
+			return handleShutdown(&handler)
+		}),
 	)
 	if err != nil {
 		return err
 	}
+
+	// Store provider reference in handler
+	handler.WasmcloudProvider = p
 	providerCh := make(chan error, 1)
 	signalCh := make(chan os.Signal, 1)
 
 	// Handle RPC operations
-	stopFunc, err := server.Serve(p.RPCClient, &Provider{})
+	stopFunc, err := server.Serve(p.RPCClient, &Handler{})
```

Comment:
To address https://github.com/wasmCloud/provider-sdk-go/pull/14#discussion_r1626791072, how about:
```suggestion
	stopFunc, err := server.Serve(p.RPCClient, &Handler{
   WasmcloudProvider: p,
})
```

Reply:
Would this be something we can autogenerate? I did try to do something like this at first, however the handler interface is generated from wit-bindgen-wrpc so I couldn't put it in the provider-sdk-go as it was before

Reply:
The other downside is from the code just a bit above, where we want to pass the handler to the callback functions for new links / health checks / shutdown logic

Reply:
@brooksmtownsend Rebase and pull in the changes from #19. It handles passing the `Provider` down into the server.

Reply:
I did 😄